src/widgets/view/view.rs

```diff
@@ -117,13 +119,18 @@ impl View {
         }
         let cursor_line = self.cursor.line - self.cache.before();
         let nb_lines = self.cache.lines().len() as u64;
+
+        self.gutter_size = 1;
+        for _ in (self.cache.before()+nb_lines+self.cache.after()).to_string().chars() {
```

Comment:
Is this just getting the length of the string representation of (self.cache.before() + nb_lines + self.cache.after())? If so, perhaps something like
```rust
self.gutter_size += (self.cache.before() + nb_lines + self.cache.after()).to_string().len();
```
might be more appropriate?

Reply:
I've actually unsure if doing this in xi-term is best. Wouldn't it make sense to add a file_line_count() function in Xrl::LineCache and then just calling from xi-term?

And your right that is much more efficient than how i was doing it, i have a nack for woorking on xi-term way to early in the morning :sleepy:

Reply:
That sounds reasonable to me.

Reply:
I decided to go with your method. after all, mostly because i'm lazy